src/main/java/Duke.java

```diff
@@ -1,10 +1,40 @@
+import Duke.Tool.*;
+import Duke.Task.*;
+
+/**
+ * Represent Duke, a task scheduling bot.
+ */
 public class Duke {
+    
+    private Storage storage;
+    private TaskList taskList;
+    private Ui ui;
+    private Parser parser;
+    private Command command;
+
+    /**
+     * Construct a new Duke object.
+     * @param filePath path of file storing the data.
+     */
+    public Duke(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        parser = new Parser();
+        command = new Command();
+        taskList = new TaskList(storage.load());
+    }
+
+    /**
+     * Execute the bot.
+     */
+    public void run() {
+        this.ui.printWelcomeMessage();
+        parser.parse(storage, ui, taskList, command);
```

Comment:
Maybe we could standardize the use of `this` keyword as these statements imply there's another ui/parser object?

Reply:
> There are some minor coding violations that can be easily fixed! Overall, well done!

Thank you for all the helpful feedbacks!! I will work on them! 👍